### ISSUE TITLE
stopPropagation

### ISSUE THREAD

**Issue by moebrowne** [joestrong/heatmapper#5]
If there are event handlers on elements in the page that call `event.stopPropagation()` the click never reaches the click handler you add to the document and so isn't recorded.

This can be seen in the example document by clicking the 'hamburger' menu in the top right

**Comment by joestrong** [comment]
Yeah, I did notice this, though not sure on how to fix. Open to any suggestions

**Comment by moebrowne** [comment]
A good 'catch all' way to go might be to use some kind of invisible overlay which clicks can be fired on and then using `document.elementFromPoint(X,Y)` you can get the element, would just need to pass the click through somehow...

**Comment by joestrong** [comment]
I've not come across `document.elementFromPoint(X,Y)` before, that's really cool. I suppose an overlay would get around the non-bubbling issue. Probably needs a try. An overlay might block hover effects though.

**Comment by moebrowne** [comment]
I've not tested it but what about overloading the `stopPropagation` method?

``` js
var oldStopPropogation = Event.prototype.stopPropagation;

Event.prototype.stopPropagation = function(e) {
    ClickTracker.placeClick(e);
    oldStopPropogation()
}
```

Not 100% sure if calling `oldStopPropogation` is going to create a circular reference though...

**Comment by moebrowne** [comment]
After reading [this](http://stackoverflow.com/a/7259956) article on StackOverflow seems to agree that this is something we can do, they used slightly different code:

``` js
var overriddenStop =  Event.prototype.stopPropagation;

Event.prototype.stopPropagation = function(){
    ClickTracker.placeClick(e);
    overriddenStop.apply(this, arguments);
}
```

**Comment by joestrong** [comment]
Cool! This is a good idea!

**Comment by joestrong** [comment]
Have implemented this here: https://github.com/joestrong/heatmapper/pull/16
